makefile

```diff
@@ -55,6 +55,19 @@ STACK_FLAGS := $(STACK_FLAGS) --flag dex:llvm-head
 STACK := $(STACK) --stack-yaml=stack-llvm-head.yaml
 endif
 
+CLANG = $(shell for clangversion in $(possible-clang-locations) ; do \
+	if [[ $$(command -v "$$clangversion" 2>/dev/null) ]]; \
+	then echo "$$clangversion" ; break ; fi ; done)
+possible-clang-locations = clang++-9 clang++-10 clang++-11 clang++
+clang-version-compatibile = $(shell $(CLANG) -dumpversion | awk '{ print(gsub(/^((9\.)|(10\.)|(11\.)).*$$/, "")) }')
+check-clang-version:
+ifeq (, $(CLANG))
+	$(error "Please install clang++-9")
+endif
+ifneq ($(clang-version-compatibile), 1)
+	$(error "Please install clang++-9")
+endif
```

Comment:
Are those ifs really part of the `check-clang-version` rule? I thought that the recipe of a single target always has to be indented by tabs. If they're not then we might be able to remove that target.

Reply:
Oh and also please make sure that we _don't_ check this when the env variable [`DEX_LLVM_HEAD`](https://github.com/google-research/dex-lang/blob/main/makefile#L50-L56) is set.

Reply:
I believe that the ifs are part of the rule (There's some conversation on this [here](https://unix.stackexchange.com/questions/220447/checking-environment-variables-value-in-makefile), and I also looked at it [here](https://gist.github.com/AlexDikelsky/f5e498c4a5257f4ab12838dfa7a9dd61)).

The reason I have the rule in the first place is so that you can still run `make clean` without having the right clang version, but I'm not sure if that is important. In [3589d7f](https://github.com/google-research/dex-lang/pull/639/commits/3589d7fc4648011990c48b50cbb537e1d2bbd5ec) I removed the rule, but I think adding it back in would be fairly simple.

Reply:
Today I learned! Thanks for investigating this.

Reply:
Adding the rule sounds good to me and might be nicer. This looks ready to merge, but I can wait for you to add that if you'd prefer. Just let me know.

Reply:
Thanks!

Before this is merged, I want to change the `CLANG :=` to respect `DEX_LLVM_HEAD`, since I think it will try to use clang++-9 even if that is set.